articles/140_topic_and_service_name_mapping.md

```diff
@@ -265,68 +231,56 @@ The standard reserves the right to use up to 8 characters for the prefix in case
-The actual length of a partition field may be limited to 256 characters, however this varies drastically depending on the vendor.
-RTI Connext does not allow a creation of a publisher/subscription with a partition length of 248 characters, whereas FastRTPS does not have any limitation in length.
-Please bare in mind, that the length of the partition gets further diminished due to the introduction of a ROS specific prefix.
-The actual length of a ROS Topic, including the namespace hierarchy and the base name of the topic, may thus be varying in length as well.
-Yet, the base name token must not exceed the length of 256 characters as this is getting mapped directly as the DDS topic.
+While testing our implementation with Connext, we encountered some additional limitations when it comes to the topic length.
+That is, for example the length of a service name has tighter limits than the length of the ROS Topics.
+The RTI Connext implementation for service names are suffixed with the GUID value of the DDS participant for the service response topic.
+Additionally, a content filtered topic (max length 256 characters) is created which is mapped from the suffixed service name.
+Therefore when linking against `rmw_connext_c` or `rmw_connext_cpp`, service names cannot be longer than 185 characters including the namespace hierarchy and any ros specific prefixes.
 
 ### Communicating with Non-ROS Topics
 
 Since all ROS topics are prefixed when being converted to DDS topic names, it makes it impossible to subscribe to existing DDS topics which do not follow the same naming pattern.
 For example, if an existing DDS program is publishing on the `image` topic (and is using the DDS equivalent to the ROS message type) then a ROS program could not subscribe to it because of the name mangling produced by the implicit ROS specific namespace.
 Therefore to allow ROS programs to interoperate with "native" DDS topic names the API should provide a way to skip the ROS specific prefixing.
 
-#### Examples and Ideas for Communicating with Non-ROS Topics
 
-Note that these are not part of the proposal, but only possible solutions to the issue of communicating with "native" DDS topics.
-This section should be update when/if a complete solution is found.
+There is an option in the API, a boolean `avoid_ros_namespace_convention` in the qos_profile which can be set to `false` to use ROS prefix and `true` to not using ROS namespace prefixing.
 
-It may be possible, through an option in the API or through some name syntax, to get a similar break down but without the ROS specific prefix.
+For example:
 
-For example with an option in the API:
+| ROS Name              | avoid_ros_namespace_conventions    | DDS Topic   |
+|-----------------------|------------------------------------|-------------|
+| `rostopic://image`    | `false`                            | `rt/image`  |
+| `rostopic://image`    | `true`                             | `image`     |
```

Comment:
my understanding is that this will result in `/image` (based on discussion from https://github.com/ros2/rmw_fastrtps/pull/192#discussion_r178970046).
We can leave this as is in the design doc if we ticket the missing bit that is to pass the user-specified topic name to rmw instead of the expanded one.
Otherwise we should correct it here

Reply:
yes that's correct I think it should be `/image` instead of `image`, as of now by default the topic name being passed here will already have a `/` appended to it (if `/` is not already present) be a fully qualified as ros topic isn't it?

Reply:
https://github.com/ros2/ros2/issues/496 has been created to track what is described here but not yet implemented